crypto/limbs/limbs.inl

```diff
@@ -35,9 +35,24 @@
 #endif
 typedef uint8_t Carry;
 #if LIMB_BITS == 64
+#if !defined(OPENSSL_AARCH64)
 #pragma intrinsic(_addcarry_u64, _subborrow_u64)
 #define RING_CORE_ADDCARRY_INTRINSIC _addcarry_u64
 #define RING_CORE_SUBBORROW_INTRINSIC _subborrow_u64
+#else
+#define RING_CORE_ADDCARRY(_carry, _a, _b, _r, _ret) do \
+{ \
+  Limb tmp = (_b) + (_carry); \
+  *(_r) = (_a) + tmp; \
+  (_ret) = ((_a) > (*(_r))) | ((_b) > tmp); \
+} while (0)
+#define RING_CORE_SUBBORROW(_borrow, _a, _b, _r, _ret) do \
+{ \
+  Limb tmp = (_a) - (_borrow); \
+  *(_r) = tmp - (_b); \
+  (_ret) = ((_a) < tmp) | ((*(_r)) > tmp); \
```

Comment:
I did my best, you know the rest... Please validate this math

Reply:
AFAICT, these macros are just doing what the `#else` branches below already do, so we can just undo this change if you take my suggestion above.

Reply:
It's not possible to use the `#else` branches below unfortunately because they use 128-bit `__uint128_t` type which is not available on MSVC 🤷

Reply:
> It's not possible to use the `#else` branches below unfortunately because they use 128-bit `__uint128_t` type which is not available on MSVC 🤷

Thanks. I forgot about that. I will think about what to do. Probably it makes sense to change the `#else` branches so that they don't require `__uint128_t` and then are shared across MSVC and non-MSVC targets, if that doesn't negatively affect the performance.

Reply:
I'll leave this decision up to you if you don't mind 🙂

Reply:
There are three aspects of this:
1. Get something working. If we can rely on clang as the C compiler, then this is less much less work than what's in this PR.
2. Get it working with MSVC as the C compiler.
3. Unifying, if practical, the MSVC and non-MSVC code paths.

If we want to have, for expediency, a separate MSVC-only code path for now, then we "just" need to convert your `#define` macros into true inline functions, since we don't do that `#define` stuff except as a last resort.

Reply:
I thought I'd leave the `crypto/fipsmodule/bn/internal.h` changes as is because they're present [upstream in BoringSSL](https://github.com/google/boringssl/blob/47c5f9d2f6c294622baf07c3373de752fd518b03/crypto/fipsmodule/bn/internal.h#L418), but if you'd prefer to revert that file, please let me know 🙂